prover/src/lib.rs

```diff
@@ -80,15 +78,22 @@ pub fn prove(
 struct ExecutionProver {
     options: ProofOptions,
     num_stack_inputs: usize,
-    num_stack_outputs: usize,
+    overflow_inputs: Vec<Felt>,
+    outputs: ProgramOutputs<Felt>,
 }
 
 impl ExecutionProver {
-    pub fn new(options: ProofOptions, num_stack_inputs: usize, num_stack_outputs: usize) -> Self {
+    pub fn new(
+        options: ProofOptions,
+        num_stack_inputs: usize,
+        overflow_inputs: Vec<Felt>,
```

Comment:
I wonder if we should just pass the full list of inputs here. Then, we can use these directly in construction of public inputs (instead of reading them from the trace).

We could also add a `debug_assert` in `get_pub_inputs()` to make sure values in the trace are consistent with inputs/outputs.